applications/GeoMechanicsApplication/custom_constitutive/coulomb_yield_surface.cpp

```diff
@@ -61,6 +61,36 @@ Vector CoulombYieldSurface::DerivativeOfFlowFunction(const Vector&, CoulombAvera
     return result;
 }
 
+void CoulombYieldSurface::UpdateSurfaceProperties(double InitialFrictionAngle,
+                                                  double FrictionAngleStrengthFactor,
+                                                  double InitialCohesion,
+                                                  double CohesionStrengthFactor,
+                                                  double InitialDilatancyAngle,
+                                                  double DilatancyAngleStrengthFactor,
+                                                  double kappa)
+{
+    mFrictionAngle = this->UpdateFrictionAngle(InitialFrictionAngle, FrictionAngleStrengthFactor, kappa) *
+                     3.14159265358979323846 / 180;
+    mCohesion = this->UpdateCohesion(InitialCohesion, CohesionStrengthFactor, kappa);
+    mDilatationAngle = this->UpdateDilatancyAngle(InitialDilatancyAngle, DilatancyAngleStrengthFactor, kappa) *
+                       3.14159265358979323846 / 180;
+}
+
+double CoulombYieldSurface::UpdateFrictionAngle(double InitialFrictionAngle, double StrengthFactor, double kappa) const
+{
+    return InitialFrictionAngle + StrengthFactor * kappa;
+}
+
+double CoulombYieldSurface::UpdateCohesion(double InitialCohesion, double StrengthFactor, double kappa) const
```

Comment:
Due to the naming, I thought it might do something to members of the class (i.e. update a cohesion and store it). To me it would be clearer if this function (same with dilatency and friction angle) is called something along the lines of:
```suggestion
double CoulombYieldSurface::CalculateUpdatedCohesion(double InitialCohesion, double StrengthFactor, double kappa) const
```